File: exemploscodigos/Exemplo 00 - Exemplos em Python/stubs.py
# Exemplos de stubs.

# Importar módulos.

import interfaces

# ---------------------------------------------------------

class StubServicoAutenticacao(interfaces.IServicoAutenticacao):

    # Atributos de classe.

    __TRIGGER = 100                 # Valor a ser usado como gatilho.

    # Métodos para solicitar serviços.

    def autenticar(self, matricula, senha):
        if(matricula.get_valor() == StubServicoAutenticacao.__TRIGGER):
            return True
        else:
            return False

File: exemploscodigos/Exemplo 10 - Controladora de interacao com PDCurses/stubs.cpp
#include "stubs.h"

const string StubServicoAutenticacao::INVALIDO = "12345";

bool StubServicoAutenticacao::autenticar(CPF cpf, Senha senha){
    if(cpf.getValor().compare(INVALIDO) == 0)
        return false;
    return true;
}


